priv/templates/extended_bin

```diff
@@ -620,7 +620,7 @@ set +e
 TMP_NAME_ARG=$(awk 'function check_name(file)
 {
     # if file exists, then it should be readable
-    if (system("test -f "file) == 0 && system("test -r "file) != 0) {
+    if (system("test -f \'"file"\'") == 0 && system("test -r \'"file"\'") != 0) {
```

Comment:
with the gsub+singelquote approach, check_name can handle paths with any of
```
 -!"#$&()*+, .[]'
```
in them

however: a `"` in the path will break the extended_bin later in relx_get_nodename

Reply:
the only special characters in a POSIX path are `0x2f` (`/`, the path separator) and `0x00`. All other characters may occur at any position.

with this PR merged, the `extended_bin` script can handle any occurrence of  characters `0x01-0x2e,0x30-0xff` in path components.

however: because of [this code in erlexec.c](https://github.com/erlang/otp/blob/38f6106e04e85802638fc6d5bb869a5b8b939f07/erts/etc/common/erlexec.c#L1284)

it  will choke on single `0x22` (`"`) bytes and fail to start empd if they occur

In FreeBSD truss(1) output it looks like this:
```
86980: execve("/bin/sh",[ "sh", "-c", ""/tmp/foo/a"b/erts-11.2.2.13/bin/epmd" -daemon" ],0x800a28000) EJUSTRETURN
```
Or in Linux strace
```
[pid 122478] execve("/bin/sh", ["sh", "-c", "\"/tmp/foo/a\"b/erts-13.0/bin/epmd"...], 0x7ffce4a7eab0 /* 38 vars */ <unfinished ...>
```

I do not think that erlexec (or any of the other places where the erlang runtime does a `sh -c` with arguments that might contain `"`) needs improvement. Those who create files and directories with characters excluded from the [Portable Filename Character Set](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_282) deserve their fate.

Nevertheless i think this PR is an improvement since it fixes the only(?) place in the `extended_bin` script where  interesting characters in paths are not correctly escaped.

But i am also fine with a 'WONT FIX' outcome; extended_bin is complex enough without the PR.